src/utils/filterEpisodes.ts

```diff
@@ -0,0 +1,16 @@
+import { IEpisode } from "./IEpisode";
+
+function filterEpisodes(
+    inputVal: string,
+    listOfAllEpisodes: IEpisode[]
+): IEpisode[] {
+    return listOfAllEpisodes.filter((oneEpisode) => {
+        const summary = oneEpisode.summary as string;
+        return (
+            oneEpisode.name.toLowerCase().includes(inputVal.toLowerCase()) ||
+            summary.toLowerCase().includes(inputVal.toLowerCase())
```

Comment:
We know that `episode.summary` can be `null`, if we didn't this would be very unclear. If summary is null then line 11 `summary.toLowerCase()` will likely crash. Consider a type guard for the case that summary (or even name too) is NOT a string.